### PR TITLE
chore(api): pin dependencies

### PR DESCRIPTION
**Pull request by jooola**
- Remove unused markdown package
- Change django packages version range

### CODE REVIEW

**Comment by jooola** [comment]
Follows #1889

**Comment by paddatrapper** [comment]
We'll have to have a look at which packages are available in the various supported distros and then try ensure that we are compatible with them. I can go through them on Monday and draw up a matrix

**Comment by jooola** [comment]
> We'll have to have a look at which packages are available in the various supported distros and then try ensure that we are compatible with them. I can go through them on Monday and draw up a matrix

Shouldn't we leave this tweaking to the packaging repository ?

Hope this doesn't pull us too far back in time.

**Comment by paddatrapper** [comment]
It depends on how much code change is required to support those versions

**Comment by jooola** [comment]
I took bullseye as reference for now, older systems should pull from pip:
- https://packages.debian.org/bullseye/python3-coreapi 2.3.3 (but coreapi is deprecated as of drf 3.10, so we might want to remove it https://www.django-rest-framework.org/coreapi/schemas/)
- https://packages.debian.org/bullseye-backports/python3-django  3.2.12 (using backports)
- https://packages.debian.org/bullseye/python3-djangorestframework  3.12.1
- https://packages.debian.org/bookworm/python3-djangorestframework-spectacular 0.22.0 but only on bookworm
- https://packages.debian.org/bullseye/python3-django-filters 2.4.0
- https://packages.debian.org/bullseye/python3-psycopg2 2.8.6

File: api/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,7 @@
...
-coreapi~=2.3.3
-django~=3.2.13
-django-filter~=21.1.0
-djangorestframework~=3.13.1
-drf-spectacular~=0.22.1
-markdown
-psycopg2
+django<4.0.0,>=3.2.12
+django-filter<22.0.0,>=2.4.0
+djangorestframework<4.0.0,>=3.12.1
```
Comment (jooola) [inline api/requirements.txt:5]: I'll put a smaller range on this.

**Comment by paddatrapper** [comment]
If we require systems to install dependencies with pip, we may as well target the latest version. Also by requiring pip, we will need to deprecate the deb packages we supply as they will most likely break things on the system by installing newer python dependencies than the system expects.

**Comment by jooola** [comment]
> If we require systems to install dependencies with pip, we may as well target the latest version. Also by requiring pip, we will need to deprecate the deb packages we supply as they will most likely break things on the system by installing newer python dependencies than the system expects.

Not sure to understand. The version range should now support packages such as python3-django on bullseye, so that if someone install this deb package, during the api install, pip will pick up the already installed version and not install download a newer version.

Did you intent to add those packages to the `packages.ini` files ? I though you wanted to make your life easier during the packaging process, hence I widen the version ranges.

If we want to support buster, we will have to revert back to django 2.2, so no maybe that's not worth it. I would be against anyway.

I don't understand, we require pip in any case. Many deps aren't packaged.

I think we should explain what is our vision (short and long term, for the main and packaging repository) with regards to dependencies. Most of the time we work on it, we either don't understand each other or argue about something that isn't clear.

**Comment by paddatrapper** [comment]
For the Debian package, as of alpha.9, we install all Python dependencies aside from `rgain3` from Ubuntu repos for Bionic and Xenial. We can't really use pip for them because that would have to be part of the post-install script run as root. Thus, the pip packages will end up being installed globally which can be a problem when they override older versions of dependencies in the repos. They then end up breaking unrelated (to LibreTime) software because they are too new.

Either we need to support the versions of packaged python dependencies in the distros we support or we need to deprecate the Deb packages.

There is also a whole Debian policy issue about installing things from pip in the post-inst, but that isn't as important here

**Comment by jooola** [comment]
I would rather put energy in the development of features/upgrades than doing maintenance a package requires. I don't think we can move forward with the same lifecycle of distribution packages. They are slow and stable, we need to move faster.

I think we should deprecate the deb packages for the time being.

If a package should be maintained, it will probably have to wait that the LibreTime dependencies versions are present in the distributions. Doing it the other way around will prevent us from having this "bug fixing / water proofing" period before landing in distributions. We should keep being upstream of the packaging process, and not mix both dev and packaging.

**Comment by paddatrapper** [comment]
Considering how behind the packages are and how much work it would be to get them back up to date, I agree that deprecating them is the right thing to do. Hopefully once we have docker images available, people can use them or the Ansible role to install LT instead.

**Comment by jooola** [comment]
@paddatrapper Do you want to add something to this PR ?

**Comment by paddatrapper** [comment]
No, I think this is good

**Pull request merged**